### PR TITLE
refactor: use base token from network

### PR DESCRIPTION
**Pull request by nicole-obrien**
## Summary

> Please summarize your changes, describing **what** they are and **why** they were made.

...

## Changelog

```
- Please write a list of changes
```

### Platforms

> Please select any platforms where your changes have been tested.

-   **Desktop**
    -   [ ] MacOS
    -   [ ] Linux
    -   [ ] Windows
-   **Mobile**
    -   [ ] iOS
    -   [ ] Android

### Instructions

> Please describe the specific instructions, configurations, and/or test cases necessary to **test and verify** that your changes work as intended.

...

## Checklist

> Please tick the following boxes that are relevant to your changes.

-   [ ] I have followed the contribution guidelines for this project
-   [ ] I have performed a self-review of my own code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have added or modified tests that prove my changes work as intended
-   [ ] I have verified that new and existing unit tests pass locally with my changes
-   [ ] I have verified that my latest changes pass CI workflows for testing and linting
-   [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/mobile/components/drawers/send/views/AmountView.svelte [hunk 1]
```diff
@@ -25,20 +27,24 @@
...
-            allowedDecimals = 0
+    $: if (asset?.metadata?.standard === TokenStandard.BaseToken) {
+        if (!asset?.metadata?.useMetricPrefix) {
+            if (unit === asset?.metadata.unit) {
+                allowedDecimals = Math.min(asset?.metadata.decimals, 18)
+            } else if (unit === asset?.metadata?.subunit) {
+                allowedDecimals = 0
+            }
+        } else if (asset?.metadata?.useMetricPrefix) {
+            allowedDecimals = IOTA_UNIT_MAP?.[unit?.substring(0, 1)] ?? 0
```
Comment (maxwellmattryan) [inline packages/mobile/components/drawers/send/views/AmountView.svelte:44]: This should be its own function e.g. `getAllowedDecimals` or something

File: packages/shared/components/inputs/AssetAmountInput.svelte [hunk 2]
```diff
@@ -32,14 +34,16 @@
...
-    $: if (!asset?.metadata?.useMetricPrefix) {
-        if (unit === asset?.metadata.unit) {
+    $: if (asset?.metadata?.standard === TokenStandard.BaseToken) {
+        if (asset?.metadata?.useMetricPrefix) {
+            allowedDecimals = IOTA_UNIT_MAP?.[unit?.substring(0, 1)] ?? 0
+        } else if (unit === asset?.metadata.unit) {
             allowedDecimals = Math.min(asset?.metadata.decimals, 18)
         } else if (unit === asset?.metadata?.subunit) {
             allowedDecimals = 0
         }
```
Comment (maxwellmattryan) [inline packages/shared/components/inputs/AssetAmountInput.svelte:44]: Same as above - extract into its own function

File: packages/shared/components/inputs/UnitInput.svelte [hunk 3]
```diff
@@ -1,36 +1,40 @@
 <script lang="ts">
     import { Dropdown2 } from 'shared/components'
-    import { ITokenMetadata } from '@core/wallet'
-    import { IDropdownItem, IotaUnit } from '@core/utils'
+    import { TokenMetadata, TokenStandard, getUnitFromTokenMetadata } from '@core/wallet'
+    import { IotaUnit, IDropdownItem } from '@core/utils'
```
Comment (maxwellmattryan) [inline packages/shared/components/inputs/UnitInput.svelte:4]: Feels like `IotaUnit` should be with the other token logic

File: packages/shared/lib/core/profile/actions/profiles/checkAndMigrateProfiles.ts [hunk 4]
```diff
@@ -156,6 +156,9 @@ function persistedProfileMigrationToV9(existingProfile: IPersistedProfile): void
 }
 
 function persistedProfileMigrationToV10(existingProfile: IPersistedProfile): void {
+    const network = NETWORK?.[existingProfile?.networkProtocol]?.[existingProfile?.networkType]
+    existingProfile.network = network
+
```
Comment (maxwellmattryan) [inline packages/shared/lib/core/profile/actions/profiles/checkAndMigrateProfiles.ts:161]: See comment on other PR

**Review by maxwellmattryan — CHANGES_REQUESTED** [review]
Have a few minor requested changes :pray: 

We need to cleanup the `wallet` core module (it should be in contexts anyway) as it currently stands - there's lots of code that could/should be in their own modules, like `nft`, `activity`, `token`, etc. We can do this in a separate task, but I think with the newly added token logic, it should go in the already existing `token` core module along with any denomination stuff like the `IotaUnit` type. Then the `token` module would be used by `wallet` and other places.

File: packages/shared/components/inputs/UnitInput.svelte (hunk 3 above)
Comment (nicole-obrien) [reply packages/shared/components/inputs/UnitInput.svelte:4]: Out of scope of this PR to clean up location of token functions

File: packages/shared/lib/core/wallet/utils/convertToRawAmount.ts [hunk 5]
```diff
@@ -15,14 +16,15 @@ export function convertToRawAmount(amount: string, tokenMetadata: ITokenMetadata
...
+    const isBaseToken = tokenMetadata?.standard === TokenStandard.BaseToken
+    if (isBaseToken && tokenMetadata.useMetricPrefix) {
         return Big(amount * IOTA_UNIT_MAP?.[unit?.substring(0, 1)] ?? 0)
     } else {
-        if (unit && unit === tokenMetadata.unit) {
+        if (unit && isBaseToken && unit === tokenMetadata.unit) {
             const decimals = Math.min(tokenMetadata.decimals, MAX_SUPPORTED_DECIMALS)
             return Big(amount).mul(Big(10).pow(decimals))
-        } else if (unit === tokenMetadata.subunit) {
+        } else if (isBaseToken && unit === tokenMetadata.subunit) {
```
Comment (begonaalvarezd) [inline packages/shared/lib/core/wallet/utils/convertToRawAmount.ts:27]: we dont have any return here for when `!isBaseToken`, we could just `return Big(amount) `

**Review by begonaalvarezd — CHANGES_REQUESTED** [review]
Works great except for that mini comment I added 🙏🏼 
Very nice work 💪🏼

**Review by begonaalvarezd — APPROVED** [review]
🚀 works

**Pull request merged**